patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -177,6 +177,39 @@
  			Main.rand = new UnifiedRandom((int)DateTime.Now.Ticks);
  			WorldFile.LoadWorld(Main.ActiveWorldFileData.IsCloudSave);
  			if (loadFailed || !loadSuccess) {
+@@ -2596,6 +_,32 @@
+ 		public static Task serverLoadWorld() => Task.Factory.StartNew(serverLoadWorldCallBack);
+ 
+ 		public static void clearWorld() {
++
++			//Large World Enabler stuff.
++			if (lastMaxTilesX < Main.maxTilesX || lastMaxTilesY < Main.maxTilesY) {
++
```

Comment:
Downsizing should also happen (without going lower than the default 8400x2400 size though.)
Loading a super large world for 1 second shouldn't permanently impact memory usage.